src/network/protocol/messages/submitorder.zig

```diff
@@ -0,0 +1,70 @@
+const std = @import("std");
+const protocol = @import("../lib.zig");
+const Sha256 = std.crypto.hash.sha2.Sha256;
+
+pub const SubmitOrderMessage = struct {
+    version: u32,
+    order_id: [32]u8,
+    item_type: [16]u8,
+    item_amount: u64,
+    payment_type: [16]u8,
+    payment_amount: u64,
+    expiration_time: u64,
+
+    pub fn name() *const [12]u8 {
+        return protocol.CommandNames.SUBMITORDER;
+    }
+
+    pub fn serialize(self: *const SubmitOrderMessage, allocator: std.mem.Allocator) ![]u8 {
```

Comment:
do it as we usualy do. SerializeToWriter

Reply:
Okay boss